barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -254,6 +271,16 @@ template <typename Curve> class ShpleminiVerifier_ {
         commitments.emplace_back(g1_identity);
         scalars.emplace_back(constant_term_accumulator);
 
+        if (!libra_univariate_evaluations.empty()) {
```

Comment:
Is there a more clarifying condition that could be used here? e.g is this always empty for non-zk flavors and always non-empty for zk flavors?

Reply:
exactly, but it's handled somewhat manually because the PCS doesn't know anything about the Flavor. added a clarifying comment

Reply:
this is probably an indication that the PCS now needs to be templated on Flavor instead of just Curve

Reply:
agree. I think I'll stop adding stuff to Shplemini quite soon and then we'll have to refactor it, because its interface is getting quite messy